urbs.py

```diff
@@ -622,6 +720,29 @@ def res_process_capacity_rule(m, sit, pro):
             m.cap_pro[sit, pro],
             m.process.loc[sit, pro]['cap-up'])
 
+# power connection capacity: Sell == Buy
+def res_sell_buy_symmetry_rule(m, sit_in, pro_in, coin):
+# constraint only for sell and buy processes
+# and the processes musst be in the same site
+    if coin in m.com_buy:
+        pro_output_tuples = list(m.pro_output_tuples.value)
+        pro_input_tuples = list(m.pro_input_tuples.value)
+        # search the output commodities for the "buy" process
+        # out = (site,output_commodity)
+        out = set([(x[0],x[2]) for x in pro_output_tuples if x[1] == pro_in])
+        # search the sell process for the output_commodity from the buy process
+        inp_p = ([x for x in pro_output_tuples if x[2] in m.com_sell])
+        for k in range(len(inp_p)):
+            pro_out = inp_p[k][1]
+            inp = set([(x[0],x[2]) for x in pro_input_tuples if x[1] == pro_out])
+            # check: buy - commodity == commodity - sell; for a site
+            if not(inp.isdisjoint(out)):
+                return (m.cap_pro[sit_in, pro_in] ==
+                        m.cap_pro[sit_in, pro_out])
```

Comment:
All from line 728 to 741 should *probably* be moved to where `pro_output_tuples` and `pro_input_tuples` are prepared themselves. Alternatively, it might be put to a helper function similar to `commodity_subset_list` and then called. Especially variable `inp_p` and `inp` must get more telling names.